### PR TITLE
Alert PR3: add tests, more icons and final export

### PR DESCRIPTION
**Pull request by divyanshu013**
Adds `Alert` component tests, all remaining icons and the final export.

> **Note**
>
> Lot of extra diff from added icons markup and the snapshots 😅 

## Screenshots

Web:

| Information | Notice |
| - | - |
| image | image |

| Negative | Positive |
| - | - |
| image | image |

Native:

| Information | Notice |
| - | - |
| image | image  |

## Task list

In PR1 https://github.com/razorpay/blade/pull/654:
- [x] Web base functionality

In PR2 https://github.com/razorpay/blade/pull/666:

- [x] Native
- [x] More stories

In this PR (PR3):
- [x] Tests
- [x] More icons
- [x] Expose final export for `Alert`
- [x] Changelog

Plus some review comments from PR2

### CODE REVIEW

File: packages/blade/src/components/Icons/AlertIcon/index.ts [hunk 1]
```diff
@@ -1 +0,0 @@
-export { default } from './AlertIcon';
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Icons/AlertIcon/index.ts:1]: Wouldn't this be a breaking change? If someone is using this their imports will fail now

File: packages/blade/src/components/Icons/AlertTriangleIcon/AlertTriangleIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import type { ReactElement } from 'react';
+import { Svg, Path } from '../_Svg';
+import type { IconProps } from '..';
+import useIconProps from '../useIconProps';
+
+const AlertTriangleIcon = ({ size, color }: IconProps): ReactElement => {
+  const { height, width, iconColor } = useIconProps({ size, color });
+
+  return (
+    <Svg width={width} height={height} viewBox="0 0 20 20" fill="none">
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Icons/AlertTriangleIcon/AlertTriangleIcon.tsx:10]: Why is your viewBox 20 20? Its 24 24 for every icon on figma. Are you copying the correct SVG? When I tried it, the results are quite different
Screenshot 2022-08-26 at 11 44 50 AM

File: packages/blade/src/components/Alert/__tests__/Alert.native.test.tsx [hunk 3]
```diff
@@ -1,5 +1,73 @@
+import { fireEvent } from '@testing-library/react-native';
+
+import { Alert } from '..';
+import renderWithTheme from '~src/_helpers/testing/renderWithTheme.native';
+
 describe('<Alert />', () => {
-  it.skip('should render', () => {
-    // todo
+  it('should render', () => {
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Alert/__tests__/Alert.native.test.tsx:7]: Should we add snapshot checks for all the intents, borderless, & full width? Don't see any tests for those

File: packages/blade/src/components/Icons/AlertIcon/index.ts (hunk 1 above)
Comment (divyanshu013) [reply packages/blade/src/components/Icons/AlertIcon/index.ts:1]: Should be fine, this isn't merged to `master` yet. I introduced this in PR1 and later realized the icon name is different 😄

File: packages/blade/src/components/Alert/Alert.tsx [hunk 4]
```diff
@@ -111,10 +111,10 @@ const CloseButtonWrapper = isReactNative ? Box : Fragment;
 
 // todo: add all icons
```
Comment (nashcheez) [inline packages/blade/src/components/Alert/Alert.tsx:112]: I think this should be an exhaustive list of components, remove the todo comment in that case?

File: packages/blade/src/components/Alert/__tests__/Alert.native.test.tsx (hunk 3 above)
Comment (nashcheez) [reply packages/blade/src/components/Alert/__tests__/Alert.native.test.tsx:7]: +1, Should at least add tests for the two different variants - borderless & fullWidth.

File: packages/blade/src/components/Alert/Alert.tsx (hunk 4 above)
Comment (divyanshu013) [reply packages/blade/src/components/Alert/Alert.tsx]: Fixed in https://github.com/razorpay/blade/pull/670/commits/acafaadda2ed45b4dadc991fe94dcab8dfb4cad9

File: packages/blade/src/components/Alert/__tests__/Alert.native.test.tsx (hunk 3 above)
Comment (divyanshu013) [reply packages/blade/src/components/Alert/__tests__/Alert.native.test.tsx:7]: I feel it creates too many snapshots to keep sane 😅, added couple of more variations in https://github.com/razorpay/blade/pull/670/commits/e98af4b3765a48d2ed6c84ab8387b32c441f6778

File: packages/blade/src/components/Icons/AlertTriangleIcon/AlertTriangleIcon.tsx (hunk 2 above)
Comment (divyanshu013) [reply packages/blade/src/components/Icons/AlertTriangleIcon/AlertTriangleIcon.tsx]: Thanks for the 🦅 👁️ review. Updated in https://github.com/razorpay/blade/pull/670/commits/569606161204170a8bc2f730c5e7a8f5dde139aa

Also tried out a SVGR config which works well for copy pasta. [Playground url](https://react-svgr.com/playground/?expandProps=none&native=true&replaceAttrValues=%23132644%3D%7BiconColor%7D&svgProps=width%3D%7Bwidth%7D%2Cheight%3D%7Bheight%7D&svgo=false)

**Review by chaitanyadeorukhkar — APPROVED** [review]

File: packages/blade/src/components/Alert/__tests__/Alert.web.test.tsx [hunk 5]
```diff
@@ -1,5 +1,106 @@
...
+          secondary: { text: 'Link', onClick: onClickSecondary },
+        }}
+      />,
+    );
+    const primaryAction = getByText('Primary');
+    const secondaryAction = getByText('Link');
+    await user.click(primaryAction);
+    await user.click(secondaryAction);
+
+    expect(onClickPrimary).toHaveBeenCalledTimes(1);
```
Comment (anuraghazra) [inline packages/blade/src/components/Alert/__tests__/Alert.web.test.tsx:58]: Do we pass any events or something to onClick? if so let's test it.

**Review by anuraghazra — APPROVED** [review]
Mostly looks fine.

File: packages/blade/src/components/Alert/__tests__/Alert.web.test.tsx (hunk 5 above)
Comment (divyanshu013) [reply packages/blade/src/components/Alert/__tests__/Alert.web.test.tsx:58]: We just pass the callback to `BaseButton` and `BaseLink` which already have their tests for this. We don't have any usecase for events here though

**Pull request merged**